src/cdk-experimental/column-resize/overlay-handle.ts

```diff
@@ -0,0 +1,207 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {AfterViewInit, ElementRef, OnDestroy, NgZone} from '@angular/core';
+import {coerceCssPixelValue} from '@angular/cdk/coercion';
+import {Directionality} from '@angular/cdk/bidi';
+import {ESCAPE} from '@angular/cdk/keycodes';
+import {CdkColumnDef} from '@angular/cdk/table';
+import {fromEvent, ReplaySubject} from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  mapTo,
+  pairwise,
+  startWith,
+  takeUntil,
+} from 'rxjs/operators';
+
+import {_closest} from '@angular/cdk-experimental/popover-edit';
+
+import {HEADER_CELL_SELECTOR} from './selectors';
+import {ColumnResizeNotifierSource} from './column-resize-notifier';
+import {HeaderRowEventDispatcher} from './event-dispatcher';
+import {ResizeRef} from './resize-ref';
+
+// TODO: Take another look at using cdk drag drop. IIRC I ran into a couple
+// good reasons for not using it but I don't remember what they were at this point.
+/**
+ * Base class for a component shown over the edge of a resizable column that is responsible
+ * for handling column resize mouse events and displaying any visible UI on the column edge.
+ */
+export abstract class ResizeOverlayHandle implements AfterViewInit, OnDestroy {
+  protected readonly destroyed = new ReplaySubject<void>();
```

Comment:
I don't see why this needs to be a `ReplaySubject`. Since it only ever emits once and completes, it can be a regular `Subject`.

Reply:
It's a safety precaution that I've adopted as my default practice - if something subscribes after this is destroyed (perhaps due to some async event or timer) then ReplaySubject will cause it to immediately unsubscribe whereas a normal subject never would.